jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -182,13 +187,17 @@ ecma_builtin_json_parse_string (ecma_json_token_t *token_p) /**< token argument
         }
         case LIT_CHAR_LOWERCASE_U:
         {
+          if ((end_p - current_p < ECMA_JSON_HEX_ESCAPE_SEQUENCE_LENGTH + 1)) {
+            return;
+          }
+
           ecma_char_t code_unit;
           if ((end_p - current_p >= 2) && !(lit_read_code_unit_from_hex (current_p + 1, 4, &code_unit)))
```

Comment:
You should use ECMA_JSON_HEX_ESCAPE_SEQUENCE_LENGTH here too.

Reply:
Oh whoops, I intended to remove that condition there. It would be redundant otherwise.